packages/client/hmi-client/src/utils/color-schemes.ts

```diff
@@ -13,3 +14,33 @@ export const strataTypeColors = [
 	'#ffff99',
 	'#b15928'
 ];
+
+const nodeTypeColorMap = ref<{ [id: string]: string }>({});
+const count = ref(0);
+
+function getNodeTypeColor(id: string): string {
+	if (!id) {
+		return '#FFF';
+	}
+	const color = nodeTypeColorMap.value[id];
+	if (!color) {
+		setNodeTypeColor(id);
+	}
+	return nodeTypeColorMap.value[id];
+}
+
+function setNodeTypeColor(id: string): void {
+	if (!id) {
+		return;
+	}
+	nodeTypeColorMap.value[id] = nodeTypeColors[count.value];
+	if (count.value >= nodeTypeColors.length) {
+		count.value = 0;
+	} else {
+		count.value++;
+	}
+}
```

Comment:
I don't know why you want to have reactive stuff here,  but it seems like you are trying to create a categorical scale here. I think there is an easier way.

```
const scale = d3.scaleOrdinal(nodeTypeColors).domain(['a', 'b', 'c', 'd', 'e'])

scale('a') // '#a6cee3'
scale('c') // '#b2df8a'
scale('d') // '#33a02c'

```